### DISCUSSION TITLE
Updating html element with function

### DISCUSSION THREAD

**Discussion by fdinhSG** [zauberzeug/nicegui#784] [Q&A] [answered]
### Question

Take this example:

```python
from nicegui import ui, app

def click_button():
    htmltext.content = ui.html('You clicked!')

with ui.column():
    ui.button('CLICK ME', on_click=lambda: click_button())
    htmltext = ui.html('Hi')
    htmltext.content = 'Hello'

ui.run()
```

I want to modify the value of `htmltext`, and I can do it without issues accessing its content (like shown in line 9).

However, when I try to do the same in a function that I bind to a button, this doesn't work anymore: instead, a new html element is created every time the button is clicked. [What I see](https://user-images.githubusercontent.com/128691970/232546588-c839a247-a1c7-475c-af94-37f05b9be403.png).

Is this an intended effect? How can I make the function work properly?

Thanks!

**Comment by falkoschindler** [comment, marked as answer]
The line

```py
htmltext.content = ui.html('You clicked!')
```

looks strange. Try this instead:

```py
htmltext.content = 'You clicked!'
```

**Reply by fdinhSG** [reply]
Duh, that was a stupid mistake. Thank you so much!

**Reply by falkoschindler** [reply]
No worries. Mistakes happen. 😀
